### DISCUSSION TITLE
Granular permissions - working as expected?

### DISCUSSION THREAD

**Discussion by ghubz** [WeblateOrg/weblate#6011] [General]
Hi,

I have created custom access levels with groups/roles. However I see that many options are given to the users which ideally shouldn't be there. There is a group with only the following permissions in their custom role:

- POST COMMENT
- EDIT STRINGS
- ADD SUGGESTION

When I login as a user who is only part of that group (removed from the default user/viewer groups), I can also:
- Download translation memory
-  View Insights (History, Download statistics, Translation Reports)
- Download translation files as ZIP
- Search + Search and replace, Data exports
- Share
- Can see (but not edit) all languages even though I have excluded them in the Group

Is that working as expected? If not, what am I missing?

Thanks in advance

**Comment by nijel** [comment]
Yes, this is expected. Once the user has access to a component, he can see all translations in it.

> Download translation memory

The user can always manage own translation memory - strings which the user has translated or uploaded.

> Translation Reports

The report contains only own contributions in this case.

See https://docs.weblate.org/en/latest/admin/access.html for more details.

**Reply by ghubz** [reply]
Ok thanks. I might look into removing the menus from the template if a user is part of that group (if that is possible)
